### ISSUE TITLE
🐛 Auth provider trying to reuse refresh tokens causing a refresh failure

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#8845]
## 🐛 Bug

We've had reports of users being kicked out of the app unexpectedly.  I believe the auth provider is not storing the the refresh token from the a refresh response.  When the next refresh is attempted it trying to reuse the old refresh token and the refresh is failing.

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/238af5f7-4bb5-4e49-9081-33dc3236d4bd)

## 🦋 Expected Behaviour

The auth provider should store the new refresh token from a refresh and use that for the next refresh

## 📋 Steps to Reproduce

> [!IMPORTANT]
This can only be reproduced with SIC.  It does not happen with mock auth.

1. Log in, note the refresh token (A)
2. Wait for the access token to expire, observe that refresh token (A) is used to refresh and a new refresh token (B) is provided
3. Wait for the access token to expire, observe that refresh token (A) is reused and the refresh failes

## ✅ Acceptance Criteria

- [ ] User can refresh their session multiple times consecutively with SIC
